src/interface/swarm_default_names.hpp

```diff
@@ -0,0 +1,37 @@
+//========================================================================================
+// (C) (or copyright) 2020-2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef INTERFACE_SWARM_DEFAULT_NAMES_HPP_
+#define INTERFACE_SWARM_DEFAULT_NAMES_HPP_
+
+#include <string>
+#include <utility>
+
+#include "swarm_pack.hpp"
+
+#define SWARM_VARIABLE(type, ns, varname)                                                \
```

Comment:
Should this macro be advertised in the docs and used downstream?

Reply:
:+1: In general we need some documentation for sparse packing and features as a whole.